jcl/src/jdk.management/share/classes/openj9/lang/management/internal/OpenJ9DiagnosticsMXBeanImpl.java

```diff
@@ -0,0 +1,423 @@
+import java.lang.reflect.Method;
+import java.lang.reflect.Field;
+import javax.management.MalformedObjectNameException;
+import javax.management.ObjectName;
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+/*[IF !Sidecar19-SE]
+import com.ibm.jvm.Dump;
+import com.ibm.jvm.InvalidDumpOptionException;
+import com.ibm.jvm.DumpConfigurationUnavailableException;
+import com.ibm.java.lang.management.internal.ManagementPermissionHelper;
+/*[ENDIF]*/
+
+import openj9.lang.management.OpenJ9DiagnosticsMXBean;
+import openj9.lang.management.InvalidOptionException;
+import openj9.lang.management.ConfigurationUnavailableException;
+
+/**
+ * Runtime type for {@link OpenJ9DiagnosticsMXBean}.
+ * <p>
+ * Implements functionality to dynamically configure dump options and to trigger supported dump agents.
+ * </p>
+ */
+public final class OpenJ9DiagnosticsMXBeanImpl implements OpenJ9DiagnosticsMXBean {
+	private final static OpenJ9DiagnosticsMXBeanImpl instance;
+
+/*[IF Sidecar19-SE]*/
+	private static final Method dump_setDumpOptions;
+	private static final Method dump_resetDumpOptions;
+	private static final Method dump_triggerDump;
+	private static final Method dump_JavaDump;
+	private static final Method dump_HeapDump;
+	private static final Method dump_SystemDump;
+	private static final Method dump_SnapDump;
+	private static final Method dump_javaDumpToFile;
+	private static final Method dump_heapDumpToFile;
+	private static final Method dump_systemDumpToFile;
+	private static final Method dump_snapDumpToFile;
+	private static final Field managementPermission;
+	private static final Class<?> invalidDumpOptionExClass;
+	private static final Class<?> dumpConfigurationUnavailableExClass;
+/*[ENDIF]*/
+
+	static {
+/*[IF Sidecar19-SE]*/
+		try {
+/*[IF Sidecar19-SE-OpenJ9]*/
+			Module openj9_jvm = ModuleLayer.boot().findModule("openj9.jvm").get(); //$NON-NLS-1$
+			Module java_management = ModuleLayer.boot().findModule("java.management").get(); //$NON-NLS-1$
+/*[ELSE]
+			Module openj9_jvm = Layer.boot().findModule("openj9.jvm").get(); //$NON-NLS-1$
+			Module java_management = Layer.boot().findModule("java.management").get(); //$NON-NLS-1$
+/*[ENDIF]*/
+			Class<?>[] classes = AccessController.doPrivileged((PrivilegedAction<Class<?>[]>)
+				() -> new Class[] {
+					Class.forName(openj9_jvm, "com.ibm.jvm.Dump"), //$NON-NLS-1$
+					Class.forName(openj9_jvm, "com.ibm.jvm.InvalidDumpOptionException"), //$NON-NLS-1$
+					Class.forName(openj9_jvm, "com.ibm.jvm.DumpConfigurationUnavailableException"), //$NON-NLS-1$
+					Class.forName(java_management, "com.ibm.java.lang.management.internal.ManagementPermissionHelper") //$NON-NLS-1$
+				});
+
+
+			Class<?> dumpClass = classes[0];
+			invalidDumpOptionExClass = classes[1];
+			dumpConfigurationUnavailableExClass = classes[2];
+			Class<?> managementPermissionHelperClass = classes[3];
+
+			managementPermission= managementPermissionHelperClass.getField("MPCONTROL"); //$NON-NLS-1$
```

Comment:
I don't think access to `ManagementPermissionHelper.MPCONTROL` needs to be via reflection. That package is exported to this module so it should be able to use the class and field directly in all Java versions. That change can wait for a separate PR though.